src/views/design/ComponentDetail/index.tsx

```diff
@@ -599,13 +598,38 @@ const ComponentDetail = withStyles(styles)(
                     isCreateParameter={isAddingParameter}
                     onEdit={(newParameter) => {
                         const newParameters = [...newComponentDetail.parameters];
+                        console.log(`newParameters: ${JSON.stringify(newParameters)}`);
+                        console.log(`newParameter: ${JSON.stringify(newParameter)}`);
                         if (!isAddingParameter) {
                             newParameters[editParameterIndex] = newParameter;
                         }
+                        const parameters = newParameters
+                            ? newParameters
+                                .map((p) => ({
+                                    name: p.name,
+                                    value: p.value,
+                                    mandatory: matchingComponentType
+                                        ? matchingComponentType.parameters
+                                            .some((type) => type.name === p.name && type.mandatory === true)
+                                        : false,
+                                }))
+                            : [];
+                        const mandatoryParameters = parameters
+                            .filter((p: any) => p.mandatory)
+                            .sort((a: any, b: any) => a.name.toLowerCase().localeCompare(b.name));
+                        const nonMandatoryParameters = parameters
+                            .filter((p: any) => !p.mandatory)
+                            .sort((a: any, b: any) => a.name.toLowerCase().localeCompare(b.name));
+                        const parameters2 = mandatoryParameters
+                            .concat(nonMandatoryParameters)
+                            .map((p: any) => ({
+                                name: p.name,
+                                value: p.value,
+                            }));
```

Comment:
the same block of logic is used above. Extract this logic to a function and use the function in the necessary places